game/tests.py

```
@@ -3536,3 +3536,59 @@ def test_record_game_result_does_not_update_rating_or_achievements_for_pvp(self)
+        
+        res2 = self.client.get(self.detail_url)
+        self.assertEqual(res2.status_code, 200)
+        
+        res3 = self.client.get(self.detail_url)
+        self.assertEqual(res3.status_code, 429)
+        self.assertEqual(res3.json(), {"error": "Opening lookup rate limit reached. Please try again shortly."})
+
+    def test_opening_trainer_rate_limit_authenticated(self):
+        User.objects.create_user(
+            username='testuser_rl',
+            password='Password123!',
```

Ruff (0.15.18)

[error] Possible hardcoded password assigned to argument: "password"

(S106)